test/util/cert_generation.go

```diff
@@ -125,8 +128,8 @@ func defaultCATemplate() *x509.Certificate {
 		Subject:               name,
 		Issuer:                name,
 		IsCA:                  true,
-		NotBefore:             time.Now(),
-		NotAfter:              time.Now().Add(1 * time.Hour),
+		NotBefore:             clk.Now(),
+		NotAfter:              clk.Now().Add(1 * time.Hour),
```

Comment:
not your bug, but ideally, it seems crisper to calculate NotBefore and NotAfter on the same "now" timestamp.